frontend/src/components/NavBar/PrintImage/image.tsx

```diff
@@ -392,467 +185,65 @@ function DownloadImage({ classes, open, handleClose }: DownloadImageProps) {
-                options={layerDescriptionSelectorOptions}
-                setValue={val =>
-                  setToggles(prev => ({
-                    ...prev,
-                    fullLayerDescription: Boolean(val),
-                  }))
-                }
-                align="end"
-                title={t('Full Layer Description')}
-              />
-            </div>
-
-            <div
-              // disable the legend scale if the legend is not visible
-              style={{
-                opacity: legendPosition !== -1 ? 1 : 0.5,
-                pointerEvents: legendPosition !== -1 ? 'auto' : 'none',
-              }}
-            >
-              <ToggleSelector
-                value={legendScale}
-                options={legendScaleSelectorOptions}
-                setValue={setLegendScale}
-                title={t('Legend Size')}
-              />
-            </div>
-
-            <ToggleSelector
-              value={mapDimensions.width}
-              options={mapWidthSelectorOptions}
-              setValue={val =>
-                setMapDimensions(prev => ({
-                  ...(prev || {}),
-                  width: val as number,
-                }))
-              }
-              title={t('Map Width')}
-            />
-
-            <ToggleSelector
-              value={footerTextSize}
-              options={footerTextSelectorOptions}
-              setValue={setFooterTextSize}
-              title={t('Footer Text')}
-            />
-
-            <TextField
-              size="small"
-              key={defaultFooterText}
-              multiline
-              defaultValue={defaultFooterText}
-              inputProps={{ style: { color: 'black', fontSize: '0.9rem' } }}
-              minRows={3}
-              maxRows={3}
-              onChange={event => {
-                debounceCallback(setFooterText, event.target.value);
-              }}
-              variant="outlined"
-            />
-
-            <Button
-              style={{ backgroundColor: cyanBlue, color: 'black' }}
-              variant="contained"
-              color="primary"
-              className={classes.gutter}
-              endIcon={<GetApp />}
-              onClick={e => handleDownloadMenuOpen(e)}
-            >
-              {t('Download')}
-            </Button>
-            <Menu
-              anchorEl={downloadMenuAnchorEl}
-              keepMounted
-              open={Boolean(downloadMenuAnchorEl)}
-              onClose={handleDownloadMenuClose}
-            >
-              <MenuItem onClick={() => download('png')}>
-                {t('Download PNG')}
-              </MenuItem>
-              <MenuItem onClick={() => download('jpeg')}>
-                {t('Download JPEG')}
-              </MenuItem>
-              <MenuItem onClick={() => download('pdf')}>
-                {t('Download PDF')}
-              </MenuItem>
-            </Menu>
-          </div>
+          <ImagePreview
```

Comment:
Let's try to find a better way to handle props
Maybe we could use a map config, or at least create an object with shared props?

Reply:
Not a huge deal, so if you think it will just add complexity and obfuscate objects we can keep things like that for now

Reply:
I think this is a good use-case for React context — performance will be identical but we'll reduce all of the duplicate type definitions. Thanks for suggesting we improve it

Reply:
@ericboucher I added [this context provider](https://github.com/WFP-VAM/prism-app/pull/1267/files#diff-7bbedea40d81e834145c75cdb4483acd7756b0f86a5cc40ce4033b0d780780e6) to clean up the code. It might be a bit over-abstracted but it does allow us to remove over 100 lines of props + type definitions